### PR TITLE
LOG-2213: forward Container Logs to Separate Elasticsearch Indices

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* add logic to the ViaQ plugin to allow configuring elasticsearch index name based on pod annotation
* The changes are implemented as multiple commits to highlight the refactoring effort
* Allows moving the pod structured logic into the plugin from fluent config

/assign @vimalk78 @alanconway 

### Links
* https://issues.redhat.com/browse/LOG-2213

### CODE REVIEW

File: fluentd/lib/fluent-plugin-viaq_data_model/lib/fluent/plugin/filter_viaq_data_model.rb [hunk 1]
```diff
@@ -251,16 +262,19 @@ def configure(conf)
...
-            raise Fluent::ConfigError, "'static' elasticsearch_index_name configurations must define 'static_index_name'"
-          end
-          matcher = ViaqMatchClass.new(ein.tag, nil)
-          ein.instance_eval{ @params[:matcher] = matcher }
-        end
-      end
+      
+      configure_elasticsearch_index_names
+
+      # # create the elasticsearch index name tag matchers
```
Comment (alanconway) [inline fluentd/lib/fluent-plugin-viaq_data_model/lib/fluent/plugin/filter_viaq_data_model.rb:268]: delete dead code

File: fluentd/lib/fluent-plugin-viaq_data_model/lib/fluent/plugin/viaq_data_model_elasticsearch_index_name.rb [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+module ViaqDataModel
+    module ElasticsearchIndexName
+
+        def configure_elasticsearch_index_names
+            unless @elasticsearch_index_names.empty?
+                @elasticsearch_index_names.each do |ein|
```
[the remark this answers is not in the document]
Comment (jcantrill) [reply fluentd/lib/fluent-plugin-viaq_data_model/lib/fluent/plugin/viaq_data_model_elasticsearch_index_name.rb:23]: This is a module that is included into the main plugin class and defined by that class https://github.com/ViaQ/logging-fluentd/pull/25/files#diff-71fb5e1ee83dd88d119dbbb04c8b6342de38c31134ade551759d917f35dd670aL168

**Pull request merged**